src/main/deltachat/chat.ts

```diff
@@ -47,19 +47,29 @@ export default class DCChat extends SplitOut {
     chatId: number,
     name: string,
     image: string,
-    remove: number[],
-    add: number[]
+    members: number[] | null
   ) {
-    log.debug('action - modify group', { chatId, name, image, remove, add })
+    log.debug('action - modify group', { chatId, name, image, members })
     this.selectedAccountContext.setChatName(chatId, name)
     const chat = this.selectedAccountContext.getChat(chatId)
     if (chat.getProfileImage() !== image) {
       this.selectedAccountContext.setChatProfileImage(chatId, image || '')
     }
-    remove.forEach(id =>
-      this.selectedAccountContext.removeContactFromChat(chatId, id)
-    )
-    add.forEach(id => this.selectedAccountContext.addContactToChat(chatId, id))
+
+    if (members !== null) {
+      const previousMembers = [
+        ...this.selectedAccountContext.getChatContacts(chatId),
+      ]
+      const remove = previousMembers.filter(m => !members.includes(m))
+      const add = members.filter(m => !previousMembers.includes(m))
+
+      remove.forEach(id =>
+        this.selectedAccountContext.removeContactFromChat(chatId, id)
+      )
+      add.forEach(id =>
+        this.selectedAccountContext.addContactToChat(chatId, id)
+      )
+    }
```

Comment:
theoretically there is a race condition here when someone else modifies the members at the same time and ui hasn't updated yet. so I would rather see that split in the ui if that makes more sense?

Reply:
We cannot do that in the ui without a store. And there we would have the same problem. If you want to fix this, you need to do it in core, and even there it will be hard.

Reply:
> i think there are some self healing measures if the member lists are out of sync.
I don't think so

It would work if we would do it in the frontend - compare what was there when the dialog was opened and whats the resulting members after the action is done -> that way no sudden groupstate state change could result in unwanted behaviour.